src/native/libs/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -139,6 +168,36 @@ int32_t CryptoNative_EvpDigestOneShot(const EVP_MD* type, const void* source, in
     return ret;
 }
 
+int32_t CryptoNative_EvpDigestXOFOneShot(const EVP_MD* type, const void* source, int32_t sourceSize, uint8_t* md, uint32_t len)
+{
+    ERR_clear_error();
+
+    if (type == NULL || sourceSize < 0 || md == NULL)
+    {
+        return 0;
+    }
+
+    EVP_MD_CTX* ctx = CryptoNative_EvpMdCtxCreate(type);
+
+    if (ctx == NULL)
+    {
+        return 0;
+    }
+
+    int32_t ret = EVP_DigestUpdate(ctx, source, (size_t)sourceSize);
```

Comment:
The only thing I can think of for your memory corruption bug is this line, where a negative number turns into a really positive number.  We have a range checked helper called Int32ToSizeT that should probably be used instead. (Though how a negative number would get here is beyond me)

I thought we used to need to use it because the compiler forbade the cast from int32_t to size_t, but since it had already compiled I decided not to mention it earlier.